helix-term/src/ui/trailing_whitespace.rs

```diff
@@ -0,0 +1,129 @@
+use helix_view::editor::{WhitespacePalette, WhitespaceRender, WhitespaceRenderValue};
+
+use helix_core::str_utils::char_to_byte_idx;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum WhitespaceKind {
+    None,
+    Space,
+    NonBreakingSpace,
+    Tab(usize),
+    Newline,
```

Comment:
I suggest tracking them. Why? They are whitespace, and also not tracking them means that people would have the option to set in their editor configuration `newling: "trailing"` and it would not display them. It could be addressed, I'm just not sure it is worth it for me to enter there.

Reply:
Newlines are always trailing though

Reply:
Done https://github.com/helix-editor/helix/pull/7215/commits/35d6345205c7d23a97607cce83af65ffbeac37e9